controllers/extendeddaemonset_e2e_test.go

```diff
@@ -893,6 +894,160 @@ var _ = Describe("ExtendedDaemonSet e2e rollout not blocked due to already faili
+		Expect(k8sClient.Create(ctx, eds)).Should(Succeed())
+
+		eds = &datadoghqv1alpha1.ExtendedDaemonSet{}
+		Eventually(withEDS(key, eds, func() bool {
+			return eds.Status.ActiveReplicaSet != ""
+		}), timeout, interval).Should(BeTrue())
+
+		info("BeforeEach: Done creating EDS %s - active replicaset: %s\n", name, eds.Status.ActiveReplicaSet)
+	})
+
+	AfterEach(func() {
+		info("AfterEach: Destroying EDS %s\n", name)
+		eds := &datadoghqv1alpha1.ExtendedDaemonSet{}
+		Expect(k8sClient.Get(ctx, key, eds)).Should(Succeed())
+		info("AfterEach: Destroying EDS %s - canary replicaset: %s\n", name, eds.Status.Canary.ReplicaSet)
+		info("AfterEach: Destroying EDS %s - active replicaset: %s\n", name, eds.Status.ActiveReplicaSet)
+
+		Eventually(deleteEDS(k8sClient, key), timeout, interval).Should(BeTrue(), "EDS should be deleted")
+
+		pods := &corev1.PodList{}
+		listOptions := []client.ListOption{
+			client.InNamespace(namespace),
+			client.MatchingLabels{
+				datadoghqv1alpha1.ExtendedDaemonSetNameLabelKey: name,
+			},
+		}
+		Eventually(withList(listOptions, pods, "EDS pods", func() bool {
+			return len(pods.Items) == 0
+		}), timeout, interval).Should(BeTrue(), "All EDS pods should be destroyed")
+
+		erslist := &datadoghqv1alpha1.ExtendedDaemonSetReplicaSetList{}
+		Eventually(withList(listOptions, erslist, "ERS instances", func() bool {
+			return len(erslist.Items) == 0
+		}), timeout, interval).Should(BeTrue(), "All ERS instances should be destroyed")
+
+		info("AfterEach: Done destroying EDS %s\n", name)
+	})
+
+	JustAfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			eds := &datadoghqv1alpha1.ExtendedDaemonSet{}
+			Expect(k8sClient.Get(ctx, key, eds)).Should(Succeed())
+			warn("%s - FAILED: EDS status:\n%s\n\n", CurrentGinkgoTestDescription().TestText, spew.Sdump(eds.Status))
+		}
+	})
+
+	restartOnCannotStartWithinMaxSlowStart := func(configureEDS func(eds *datadoghqv1alpha1.ExtendedDaemonSet), expectedReason string) {
+		Eventually(updateEDS(k8sClient, key, configureEDS), timeout, interval).Should(
+			BeTrue(),
+			func() string { return "Unable to update the EDS" },
+		)
+
+		eds := &datadoghqv1alpha1.ExtendedDaemonSet{}
+		Expect(k8sClient.Get(ctx, key, eds)).Should(Succeed())
+		info("%s: %s - active replicaset: %s\n",
+			CurrentGinkgoTestDescription().TestText,
+			name, eds.Status.ActiveReplicaSet,
+		)
+
+		info("EDS %s - waiting for canary to not be paused\n", name)
+		eds = &datadoghqv1alpha1.ExtendedDaemonSet{}
+		Eventually(withEDS(key, eds, func() bool {
+			return eds.Status.Canary != nil
+		}), longTimeout, interval).Should(BeTrue())
+
+		pods := &corev1.PodList{}
+		listOptions := []client.ListOption{
+			client.InNamespace(namespace),
+			client.MatchingLabels{
+				datadoghqv1alpha1.ExtendedDaemonSetNameLabelKey: name,
+			},
+		}
+		Eventually(withList(listOptions, pods, "EDS pods", func() bool {
+			if len(pods.Items) < 0 {
```

Comment:
❓ 's
- Why would `len(pods.Items)` be negative in the first place? Doesn't the `len()` function always return non-negative value?
- What if `len(pods.Items) == 0` ? Should we return `false` in this case too?

Reply:
meant to make this `== 0` thanks for catching that!